docs(changelog): update changelog for tag v0.5.1 (#334)
tier4/caret_doc: docs/changelog.md

## CARET

### v0.5.1 <small>\_ Mar 11, 2024</small> {id = "0.5.1"}

> - **Update** Remove the description of response time histogram. ([caret_doc #331](https://github.com/tier4/caret_doc/pull/331))

Reviewer: ```suggestion
- **Update**:  Remove the description of response time histogram. ([caret_doc #331](https://github.com/tier4/caret_doc/pull/331))
```
Insert colons in other places as well.

Author: Fixed after `v0.4.22` in https://github.com/tier4/caret_doc/pull/334/commits/d002e71f9981ec5c1e8aa3c208c3f2cee5f822a1 .

- **Fix** Fixed that histograms are stacked, because overlapping histograms are hard to see. ([caret_analyze #475](https://github.com/tier4/caret_analyze/pull/475))

> - **Fix** Fixed problem with ros2 caret command not working when built without --symlink-install. ([ros2caret #175](https://github.com/tier4/ros2caret/pull/175))

Reviewer: ```suggestion
- **Fix** Fixed problem with ros2 caret command not working when built without `--symlink-install`. ([ros2caret #175](https://github.com/tier4/ros2caret/pull/175))
```

- **Update** Add a dynamic type check method for the plot API. ([caret_analyze #472](https://github.com/tier4/caret_analyze/pull/472))

> - **Fix** Fixed speeding up by limiting registration to init_events because id_remapper takes a long time to load measurement data. ([caret_analyze #480](https://github.com/tier4/caret_analyze/pull/480))

Reviewer: ```suggestion
- **Fix** Speeding up by limiting registration to init_events because id_remapper takes a long time to load measurement data. ([caret_analyze #480](https://github.com/tier4/caret_analyze/pull/480))
```

- **Fix** Fixed warning "More than three callbacks are registered in one subscription_handle. Skip loading callback info." when parsing certain trace data. ([caret_analyze #470](https://github.com/tier4/caret_analyze/pull/470))

> - **Fix** Fixed "filtered to 0 events" output when running ros2 caret check_ctf. ([caret_analyze #480](https://github.com/tier4/caret_analyze/pull/480))

Reviewer: This problem occurs when using filters when loading trace data (more precisely, when creating Lttng objects).
```suggestion
- **Fix**: Fixed a bug where the count of filtered data wasn't correctly displayed ("filtered to 0 events") when loading trace data. ([caret_analyze #480](https://github.com/tier4/caret_analyze/pull/480))
```
